src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -19,13 +22,82 @@ namespace Microsoft.PowerShell.Commands
     [SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly")]
     public static class JsonObject
     {
+        #region HelperTypes
+
+        /// <summary>
+        /// Context for convert-to-json operation.
+        /// </summary>
+        public readonly struct ConvertToJsonContext
```

Comment:
Can it be `ref struct`?

Reply:
It doesn't contain a reference, so why using `ref struct`?. `ref struct` makes the type a ByRef-like type, which can only exists on stack -- that means you cannot use this type and the `ConvertToJson` API in powershell script.

Reply:
I look as we use this internally - it seems a good candidate for ref struct to avoid allocations that would great for hosting scenarios. For script scenarios - why we need the API for scripts? - we could create new constructor with all these parameters.